ncm-iptables/src/main/perl/iptables.pm

```diff
@@ -530,54 +526,52 @@ RULE:	foreach $name (sort { $a <=> $b } keys %{$entries->{$table}->{rules}}) {
 #   OUTPUT: @keys     - list of keys in the right order,
 #           $?        - 0 keys sorted,
 #                     - 1 error.
-#      USE: %options_ord
+#      USE: %OPTION_SORT_ORDER
 #   ASSUME: If rule is not empty then is well formed.
-##########################################################################
 sub sort_keys {
-    my ($self,$rule) = @_;
-    my ($i, $m, $purge, $swap, $reg);
-    my (@keys, @ord);
-
-    # Check parameters.
-    if ( $rule !~ /^HASH/ ) {
-	$? = 1;
-	$@ = "bad rule";
-	return ();
-    }
+    my ($self, $rule) = @_;
 
-    @keys = keys %{$rule};
-
-    $purge = 1;
-WHILE: while( $purge ) {
-FOR:     for($i=0, $purge=0; $i<=$#keys; $i++) {
-    next if ( $keys[$i] !~ /^(err|checked)$/ );
-    splice(@keys,$i,1);
-    $purge = 1;
-    last FOR;
-}
-}
+    $self->debug(5, "Entering method sort_keys");
 
-    $swap = 1;
-    while ( $swap ) {
-	for($m=0, $swap=0; $m<$#keys; $m++) {
-	    for($i=$m+1; $i<=$#keys; $i++) {
-
-		$self->error("$keys[$i] is not a valid option\n") if ! exists $options_ord{$keys[$i]};
-		$self->error("$keys[$m] is not a valid option\n") if ! exists $options_ord{$keys[$m]};
-
-		#next
-		if (! exists $options_ord{$keys[$i]} || ! exists $options_ord{$keys[$m]}){
-		    $? = 1;
-		    $@ = "keys unsorted";
-		    return @keys;
-		}
-		next if ( $options_ord{$keys[$i]} >= $options_ord{$keys[$m]} );
-		$reg      = $keys[$i];
-		$keys[$i] = $keys[$m];
-		$keys[$m] = $reg;
-		$swap++;
-	    }
-	}
+    # Check parameters.
+    if ($rule !~ /^HASH/) {
+        $? = 1;
+        $@ = "bad rule";
+        return ();
+    }
+
+    my @keys = keys %{$rule};
+
+    my $purge = 1;
+    WHILE: while ($purge) {
+        FOR: for (my $i=0, $purge=0; $i<=$#keys; $i++) {
```

Comment:
if the code is intended to confuse people, i'd say works perfectly. i have a vague idea what is supposed to do, but i'm not sure.

```perl
my @keys = grep {$_ !~ m/^(err|checked)$/} sort keys %{$rule};
```